text/0003-blockspace-regions.md

```diff
@@ -0,0 +1,252 @@
+  5. **The solution SHOULD minimize the required advance notice for scheduling Coretime.** 
+    - Application load is highly volatile, and the core scheduling primitives should enable applications to pay for only the Coretime they need, while allowing them to gracefully handle periods of heavy load. It is quite common for applications to suddenly experience a large influx of traffic and interest which lasts for a short time.
+  6. **The solution MUST NOT allow chains to access more than their scheduled amount of Coretime.**
+  7. **The solution MUST NOT allow parachains to build up arbitrary amounts of Coretime to spend later on**
+    - The intention of regions is to ensure consistent rates of utilization by scheduled parachains. Allowing arbitrary amounts of Coretime to be built up and spent later will lead to misallocation of system resources during periods of high demand. Eliminating this type of arbitrage is necessary.
+  8. **The solution SHOULD unify all scheduling mechanisms on the relay-chain**. Maintaining multiple parallel interfaces and implementations of scheduling on the relay-chain, runtime APIs, and node-side will contribute enormously to implementation and ongoing maintenance overheads and should be avoided.
+
+## Stakeholders
+
+- Protocol Reaearchers and Developers.
+- Polkadot Parachain teams both present and future, and their users.
+- DOT Token Holders
+
+These concepts have been alluded to in [Polkadot: Blockspace over Blockchains](https://www.rob.tech/blog/polkadot-blockspace-over-blockchains/) and discussed, with previous drafts of the design, [on the Polkadot Forum](https://forum.polkadot.network/t/unifying-bulk-and-mid-term-blockspace-with-strided-regions/2228)
+
+## Explanation
+
+### Parameters:
+
+| Name                | Constant | Value     |
+| ------------------- | -------- | --------- |
+| RATE_DENOMINATOR    | YES      | 57600     |
+| SCHEDULING_LENIENCE | NO       | 16        |
+| HYPERCORES          | NO       | 8         |
+| HYPERCORES_PER_CORE | NO       | 1         |
+
+### Region and RegionSchema
+
+A region is a data structure outlined below:
+
+```rust
+// Indicates a number in the range 0..57600 (RATE_DENOMINATOR)
+type PartsPer57600 = u16;
+// Indicates a number treated as a rational over 57600 (RATE_DENOMINATOR) which
+// may be greater than 57600. This must be a u64, as regions running longer than a
+// few days would overflow a u32.
+type RationalOver57600 = u64;
+
+struct RegionSchema {
+    // Relay-chain block number at which this region becomes active.
+    start: u32,
+    // The end point of the region in relay-chain blocks, i.e. it ends at start + duration. Note that endpoints are flexible up to `SCHEDULING_LENIENCE`.
+    // This may be `None`, in the case that the true endpoint is only determined later.
```

Comment:
Via RFC-5 (#5) - the interface can be used more efficiently by not specifying the endpoint.

On the RFC-1 broker chain (or any other marketplace using that interface), you could imagine a region owner pre-emptively splitting up its coretime into e.g. 10 minute chunks, and assigning it shortly before the deadline. The `None` endpoint allows to avoid sending a message for each chunk as long as the assignment stays the same as it was before.